test/dependency_source/test_resolvelib.py

```diff
@@ -49,7 +49,7 @@ def check_deps(resolved_deps: List[Dependency], expected_deps: List[Dependency])
 def test_resolvelib():
     resolver = resolvelib.ResolveLibResolver()
     req = Requirement("flask==2.0.1")
-    resolved_deps = dict(resolver.resolve_all([req]))
+    resolved_deps = dict(resolver.resolve_all(iter([req])))
```

Comment:
Weird that all of these `iter(...)` calls are required, given that `List[...]` should already provide that type bound. I'll look into this.

Reply:
Ah, this is probably because our various interfaces take and return `Iterator`s instead of `Iterable`s. We can selectively refine those another time.